dispatches/models/nuclear_case/chgtank.py

```diff
@@ -0,0 +1,590 @@
+
+        # material holdup calculation
+        @self.Constraint(self.flowsheet().config.time,
+                          pc_set,
+                          doc="Material holdup calculations")
+        def material_holdup_calculation(b, t, p, j):
+            if (p, j) in pc_set:
+                return (
+                    b.material_holdup[t, p, j] == (
+                        b.control_volume.volume[t] *
+                        b.control_volume.phase_fraction[t, p] *
+                        b.control_volume.properties_out[t].\
+                            get_material_density_terms(p, j)))
+
+        # energy balances
+        @self.Constraint(self.flowsheet().config.time,
+                          doc="Total Enthalpy Balance")
+        def enthalpy_balances(b, t):
+            return (
+                sum(b.energy_accumulation[t, p] for p in phase_list) == (
+                    sum(b.control_volume.properties_in[t].\
+                        get_enthalpy_flow_terms(p) for p in phase_list)
+                    - sum(b.control_volume.properties_out[t].\
+                        get_enthalpy_flow_terms(p) for p in phase_list)
+                    + b.heat_duty[t]))
+
+        # energy holdup calculation
+        @self.Constraint(self.flowsheet().config.time,
+                         phase_list,
+                         doc="Energy holdup integration")
+        def energy_holdup_integration(b, t, p):
+            return b.energy_holdup[t, p] == (
+                  sum(b.material_holdup[t, p, j]
+                      for j in component_list)
+                  * b.control_volume.properties_out[t].\
+                      enth_mol_phase[p])
+
+        # computing specific heat capacity ratio (cp/cv)
+        # NOTE: using ideal gas property Cv = Cp - R
+        @self.Expression(self.flowsheet().config.time,
+                         doc="specific heat ratio, cp/cv")
+        def gama(b, t):
+            for p in phase_list:
+                if p == 'Vap':
+                    return (
+                        b.control_volume.properties_in[t].\
+                            cp_mol_phase[p]
+                        / (b.control_volume.properties_in[t].\
+                           cp_mol_phase[p]
+                           - gas_constant)
+                        )
+                else:
+                    return 1
+
+        # Add constraint to determine the tank final temperature (T out)
+        # This constraint is derived from the 1st law of thermodynamics and 
+        # Ideal gas law assuming an adiabatic process
+        # Reference: Yang (2009). Int. J. of Hydrogen Energy 34, 6712-6721.
+
+        # TO DO: add pressure out equation for an isothermal process
+
+        @self.Expression(self.flowsheet().config.time,
+                         doc="Dimensionles time")
+        def dim_less_time(b, t):
+            return (
+                self.control_volume.properties_in[t].flow_mol
+                * b.dt[t]
```

Comment:
We should think about how we should define this. One would think that we could calculate it from the defined time points (i.e. `dt[t] = t - (t-1)`). However, there are some obvious limitations to doing this. Otherwise, there is the possibility of a user defining a mismatch between dt and the actual time steps.